packages/discord.js/src/structures/BaseInteraction.js

```diff
@@ -46,7 +47,18 @@ class BaseInteraction extends Base {
      * The id of the channel this interaction was sent in
      * @type {?Snowflake}
      */
-    this.channelId = data.channel_id ?? null;
+    this.channelId = data.channel?.id ?? null;
+
+    /**
+     * The channel this interaction was sent in.
+     * @type {?TextBasedChannels|APIChannel}
+     * @private
+     */
+    this._channel =
+      (this.channelId && client.channels.cache.get(this.channelId)) ??
+      client.options.partials.includes(Partials.Channel)
+        ? data.channel
+        : null;
```

Comment:
```suggestion
```
https://github.com/discordjs/discord.js/blob/8592edba39bffb6383512d4a3b1fc5f12896cde9/packages/discord.js/src/client/actions/InteractionCreate.js#L23
Change this to `data.channel`

Reply:
I do not think this is a good idea. Only partial channel data is sent. Instead of receiving a full structure from the cache, we now get fewer data.

Reply:
`.getChannel()` handels partal structures

Reply:
After much staring at the screen, I finally understood the suggested change. Thank you!